lib/fastlane/plugin/wpmreleasetoolkit/helper/ios/ios_l10n_helper.rb

```diff
@@ -48,24 +48,33 @@ def self.strings_file_type(path:)
         #
         # @raise [RuntimeError] If one of the paths provided is not in text format (but XML or binary instead), or if any of the files are missing.
         #
-        def self.merge_strings(paths:, output_path: nil)
+        def self.merge_strings(paths:, output_path:)
           duplicates = []
           Tempfile.create('wpmrt-l10n-merge-', encoding: 'utf-8') do |tmp_file|
             all_keys_found = []
 
             tmp_file.write("/* Generated File. Do not edit. */\n\n")
-            paths.each do |input_file|
+            paths.each do |input_file, prefix|
               fmt = strings_file_type(path: input_file)
               raise "The file `#{input_file}` does not exist or is of unknown format." if fmt.nil?
               raise "The file `#{input_file}` is in #{fmt} format but we currently only support merging `.strings` files in text format." unless fmt == :text
 
-              string_keys = read_strings_file_as_hash(path: input_file).keys
+              string_keys = read_strings_file_as_hash(path: input_file).keys.map { |k| "#{prefix}#{k}" }
               duplicates += (string_keys & all_keys_found) # Find duplicates using Array intersection, and add those to duplicates list
               all_keys_found += string_keys
 
               tmp_file.write("/* MARK: - #{File.basename(input_file)} */\n\n")
               # Read line-by-line to reduce memory footprint during content copy; Be sure to guess file encoding using the Byte-Order-Mark.
-              File.readlines(input_file, mode: 'rb:BOM|UTF-8').each { |line| tmp_file.write(line) }
+              File.readlines(input_file, mode: 'rb:BOM|UTF-8').each do |line|
+                if prefix
+                  # We need to ensure the line and RegExp are using the same encoding, so we transcode everything to UTF-8.
+                  line.encode!(Encoding::UTF_8)
+                  # The `/u` modifier on the RegExps is to make them UTF-8
+                  line.gsub!(/^(\s*")/u, "\\1#{prefix}") # Lines starting with a quote are considered to be start of a key; add prefix right after the quote
+                  line.gsub!(/^(\s*)([A-Z0-9_]+)(\s*=\s*")/ui, "\\1\"#{prefix}\\2\"\\3") # Lines starting with an identifier followed by a '=' are considered to be an unquoted key (typical in InfoPlist.strings files for example)
+                end
```

Comment:
I'm not super happy with having to use RegExps here to modify the input `.strings` file, especially given one of my recurring motto while working on this modernization of our Translation Tooling was that I wanted to avoid at all cost having to parse or write `.strings` files that are in OpenStep (textual) format like here, because that format is poorly documented (no official strict spec/grammar published afaik) and can be full of edge cases. This is the whole reason why I've been using `plutil` to parse the key/value pairs of those files instead of trying to parse the file myself, and why the `.strings` files we generate with those actions use the XML output format and not the undocumented textual format.

Alas, I'm not sure there's a better solution in order to insert those prefixes in the keys (and without losing the original `/* comments */` of those strings files so that GlotPress have them too on import, hence why we need to keep the textual OpenStep format for those files we plan to make GlotPress import…).

I'm hoping the couple of [red herrings I've added to my Unit Test fixtures](https://github.com/wordpress-mobile/release-toolkit/pull/345/files#diff-8fa1a86ac87d73ab1777cffbe9502a139b2385523e2caace86d38009bf3e42d0R15) to try to trip this (and show that we didn't fall into those traps) would be enough to cover us.